lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/LaunchClass.java

```diff
@@ -41,7 +41,9 @@
 
 /**
  * This encapsulates processes started by REEF.
+ * @deprecated This class is exclusively used by the deprecated Launcher.
```

Comment:
For all these `@deprecated` messages, add the version in which it was deprecated, e.g. `@deprecated in 0.12. `